File: app/src/main/java/com/originb/inkwisenote2/modules/handwrittennotes/data/HandwrittenNotesDao.java
package com.originb.inkwisenote2.modules.handwrittennotes.data;

import androidx.room.*;

import java.util.List;
import java.util.Set;

@Dao
public interface HandwrittenNotesDao {

    @Query("SELECT * FROM handwritten_notes")
    List<HandwrittenNoteEntity> getAllHandwrittenNotes();


    @Query("SELECT * FROM handwritten_notes WHERE note_id = :noteId")
    HandwrittenNoteEntity getHandwrittenNoteForNote(Long noteId);

    @Query("SELECT * FROM handwritten_notes WHERE book_id = :bookId")
    List<HandwrittenNoteEntity> getHandwrittenNoteForBook(Long bookId);

    @Query("SELECT * FROM handwritten_notes WHERE book_id IN (:bookIds)")
    List<HandwrittenNoteEntity> getHandwrittenNoteForBooks(Set<Long> bookIds);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertHandwrittenNote(HandwrittenNoteEntity handwrittenNoteEntity);

    @Update
    int updateHandwrittenNote(HandwrittenNoteEntity handwrittenNoteEntity);

    @Update
    int updateHandwrittenNotes(List<HandwrittenNoteEntity> handwrittenNoteEntities);

    @Query("DELETE FROM handwritten_notes WHERE note_id IN (:noteIds)")
    void deleteHandwrittenNotes(Set<Long> noteIds);
}

File: app/src/main/java/com/originb/inkwisenote2/modules/handwrittennotes/data/HandwrittenNotesDao.kt
package com.originb.inkwisenote2.modules.handwrittennotes.data

import androidx.room.*

@Dao
interface HandwrittenNotesDao {
    @get:Query("SELECT * FROM handwritten_notes")
    val allHandwrittenNotes: List<HandwrittenNoteEntity?>?


    @Query("SELECT * FROM handwritten_notes WHERE note_id = :noteId")
    fun getHandwrittenNoteForNote(noteId: Long?): HandwrittenNoteEntity?

    @Query("SELECT * FROM handwritten_notes WHERE book_id = :bookId")
    fun getHandwrittenNoteForBook(bookId: Long?): List<HandwrittenNoteEntity?>?

    @Query("SELECT * FROM handwritten_notes WHERE book_id IN (:bookIds)")
    fun getHandwrittenNoteForBooks(bookIds: Set<Long?>?): List<HandwrittenNoteEntity?>?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertHandwrittenNote(handwrittenNoteEntity: HandwrittenNoteEntity?): Long

    @Update
    fun updateHandwrittenNote(handwrittenNoteEntity: HandwrittenNoteEntity?): Int

    @Update
    fun updateHandwrittenNotes(handwrittenNoteEntities: List<HandwrittenNoteEntity?>?): Int

    @Query("DELETE FROM handwritten_notes WHERE note_id IN (:noteIds)")
    fun deleteHandwrittenNotes(noteIds: Set<Long?>?)
}
